### PR TITLE
fix for vividus/issues/883

### LINKED ISSUE #883: I want "round" function might support numbers like "1.056E7b" or Vividus can correctly work with "big" numbers like 10 560 000 and alike without adding 'b'
There is a problem with numbers which have more than 5 digits (and which might look like "1.056E7b" after vividus` interpreting)
At the moment we use such with adding 'b' after it. 

Example: 
```
${x} * ${y} = ${x*y} in order to use variable ${x*y} in further calculations we add 'b' => '${x*y}b' 
132000*80 = 10 560 000 == 1.056E7
```

It would be great if mentioned numbers type will correctly collaborate with "round" functionality. 

Also I would like to notice that number in the variable '${x*y}b' (10 560 000 == 1.056E7) is sent with wrong value into field in used UI. 
Example: 
1) save expected value in '${x*y}'
2) using step 'When I enter `${x*y}b` in field located `By.id(bv1)' 
3) send request with value
it does not sent 1.056E7 but delete everything after letter (looks like 1.056) and automattically round number => 1.06
but in the same time, value of this variable is used in the possible next steps with its value = 1.056E7

### PR DESCRIPTION
**Pull request by BViktoryia**
Fixes #883

### CODE REVIEW

**Comment by BViktoryia** [comment]
fix for vividus/issues/883

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java [hunk 1]
```diff
@@ -38,8 +38,8 @@
     {
         String roundingModes = Stream.of(RoundingMode.values()).map(e -> e.name().toLowerCase())
                 .collect(Collectors.joining("|"));
-        String pattern = String.format("^round(?:\\((-?\\d+(?:\\.\\d*)?)(?:,\\s*(\\d+))?(?:,\\s*(%s))?\\))$",
-            roundingModes);
+        String pattern = String.format("^round(?:\\((-?\\d+(?:\\.\\d*)?[Ee]?-?\\+?\\d*)"
+                        + "(?:,\\s*(\\d+))?(?:,\\s*(%s))?\\))$", roundingModes);
```
Comment (uarlouski) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/expression/RoundExpressionProcessor.java:42]: please add unit and integration tests

**Review by uarlouski — COMMENTED** [review]
Commit name should reflect made changes without opening linked issue

**Comment by valfirst** [comment]
Superseded by #1018

**Pull request closed without merging**